opnode/rollup/derive/batch.go

```diff
@@ -0,0 +1,70 @@
+package derive
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/ethereum-optimism/optimistic-specs/opnode/l2"
+	"github.com/ethereum-optimism/optimistic-specs/opnode/rollup"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+// Batch format
+// first byte is type followed by bytstring.
+//
+// BatchV1Type := 0
+// BatchV1Type ++ RLP([epoch, timestamp, transaction_list]
+
+const (
+	BatchV1Type = iota
+)
+
+type BatchV1 struct {
+	Epoch        uint64
+	Timestamp    uint64
+	Transactions []hexutil.Bytes
+}
+
+type BatchData struct {
+	Epoch     rollup.Epoch // aka l1 num
+	Timestamp uint64
+	// no feeRecipient address input, all fees go to a L2 contract
+	Transactions []l2.Data
+}
+
+func ParseBatch(data []byte) (BatchData, error) {
+	var v1 BatchV1
+	if err := v1.UnmarshalBinary(data); err != nil {
+		return BatchData{}, err
+	}
+	return BatchData{Epoch: rollup.Epoch(v1.Epoch), Timestamp: v1.Timestamp, Transactions: v1.Transactions}, nil
+}
+
+// // TODO: Is this needed?
+// // EncodeRLP implements rlp.Encoder
```

Comment:
We are currently not encoding/decoding the batch in an RLP context, so it's not right now. The geth RLP package is not aware of `MarshalBinary` / `UnmarshalBinary`, so we'll need the `EncodeRLP`/`DecodeRLP` methods when e.g. RLP-encoding a list of batches. And with support for multiple batches that's exactly what I plan to do :)